packages/transport/src/modules/messages/MessageController.ts

```diff
@@ -293,20 +293,74 @@ export class MessageController extends TransportController {
         const serializedSecret = secret.serialize(false);
         const addressArray: ICoreAddress[] = [];
         const envelopeRecipients: MessageEnvelopeRecipient[] = [];
+        const peerDeletedAddressArray: string[] = [];
+        const missingOrInactiveRelationshipAddressArray: string[] = [];
         for (const recipient of parsedParams.recipients) {
             const relationship = await this.relationships.getActiveRelationshipToIdentity(recipient);
-            if (!relationship) {
-                throw TransportCoreErrors.messages.missingOrInactiveRelationship(recipient.toString());
+            if (
+                relationship &&
+                (relationship.status === RelationshipStatus.Terminated || relationship.status === RelationshipStatus.Active) &&
+                (relationship.peerDeletionInfo?.deletionStatus === "ToBeDeleted" || !relationship.peerDeletionInfo?.deletionStatus)
+            ) {
+                const cipherForRecipient = await this.secrets.encrypt(relationship.relationshipSecretId, serializedSecret);
+                envelopeRecipients.push(
+                    MessageEnvelopeRecipient.from({
+                        address: recipient,
+                        encryptedKey: cipherForRecipient
+                    })
+                );
+                addressArray.push(recipient);
             }
+            if (!(relationship && (relationship.status === RelationshipStatus.Terminated || relationship.status === RelationshipStatus.Active))) {
+                missingOrInactiveRelationshipAddressArray.push(recipient.address);
+            }
+            if (relationship?.peerDeletionInfo?.deletionStatus === "Deleted") {
+                peerDeletedAddressArray.push(recipient.address);
+            }
+        }
+        if (missingOrInactiveRelationshipAddressArray.length > 0) {
```

Comment:
looks like massive copy-paste from the other code I already commented on. Also here it is way to complex.

Reply:
I will do the whole validation here inclusive this from the 'Notification' because then it isn't anymore necessary to do it in `SendMessage.ts` and I will refactore it like you said above.

Reply:
Actually you can not do the whole validation here as the Transport library does not know Notifications or Requests.

I think it would be better to just remove the validation HERE, as transport should only transport the stuff that the runtime wants to transport. The only thing we could check here if one of the recipients is already deleted. This is a real transport issue as it will never be accepted by the backbone.